aries_vcx/src/handlers/connection/inviter/state_machine.rs

```diff
@@ -451,15 +450,14 @@ impl SmConnectionInviter {
         new_service_endpoint: String,
     ) -> VcxResult<SignedResponse> {
         request.connection.did_doc.validate()?;
-        let thread_id = &request.get_thread_id().ok_or(VcxError::from_msg(VcxErrorKind::InvalidJson, "Missing ~thread decorator field in request"))?;
-        verify_thread_id(&thread_id, &A2AMessage::ConnectionRequest(request.clone()))?;
+        verify_thread_id(&request.get_thread_id(), &A2AMessage::ConnectionRequest(request.clone()))?;
         let new_recipient_keys = vec!(new_pairwise_info.pw_vk.clone());
         Response::create()
             .set_did(new_pairwise_info.pw_did.to_string())
             .set_service_endpoint(new_service_endpoint)
             .set_keys(new_recipient_keys, new_routing_keys)
             .ask_for_ack()
-            .set_thread_id(&thread_id)
+            .set_thread_id(&request.get_thread_id())
```

Comment:
Would feel safer to always copy from `self.thread_id`, though after verification those values should be the same (except when the verification of the message thread id is forgotten or incorrect like happened above)

Reply:
I don't think it should matter at all, but will make it `self.thread_id` if you think it's better.

Reply:
Actually, `self.thread_id` can't be used here because it is undefined in the scenario where the inviter is creating a connection from received request (the thread id is set after the response is built). There are ways to satisfy your demand to use `self.thread_id`, but I don't see the upside of that.